plugin-server/src/ingestion/ingestion-consumer.ts

```diff
@@ -195,6 +199,17 @@ export class IngestionConsumer {
         logger.info('🔁', `${this.name} - stopping`)
         this.isStopping = true
 
+        // Wait for current batch and scheduled promises to complete before proceeding with shutdown (if feature flag is enabled)
+        if (this.hub.KAFKA_CONSUMER_GRACEFUL_SHUTDOWN) {
+            if (this.currentBatchPromise) {
+                logger.info('🔁', `${this.name} - waiting for current batch to complete`)
+                await this.currentBatchPromise
+            }
+
+            logger.info('🔁', `${this.name} - waiting for scheduled promises to complete`)
+            await this.promiseScheduler.waitForAll()
+        }
+
         // Mark as stopping so that we don't actually process any more incoming messages, but still keep the process alive
         logger.info('🔁', `${this.name} - stopping batch consumer`)
         await this.kafkaConsumer?.disconnect()
```

Comment:
I _thought_ this is covered by the kafkaConsumer disconnect here. It also waits for the current consumed batch which wraps the handleBatch function 🤔 

Happy to try this too but it definitely feels redundant...